### PR TITLE
Adds two Christmas Spells

### PR DESCRIPTION
**Pull request by Arthurdentist**
This PR adds two new Christmas Spells, only available on Christmas Day! 

The Spells Are:

Curse of the Reindeer, its a Christmas themed Horsemask Curse! Sprites courtesy of NigglyWiggly

reindeer

Elf Curse: Clown curse but worse. Enjoy being a miserable elf on Christmas, and every other day after that too. Turns the victim into an elf, gives them an unmovable elf hat, elf uniform and elf shoes. Sprites courtesy of Blithering

elf

:cl: 
- rscadd: Added two new spells that will be available for wizards to purchase on Christmas Day

### CODE REVIEW

File: code/modules/spells/targeted/equip/horsemask.dm [hunk 1]
```diff
@@ -37,3 +39,28 @@
...
+	hud_state = "wiz_reindeer"
+
+	animal = "reindeer"
+
+/spell/targeted/equip_item/horsemask/christmas/New()
+	..()
+	equipped_summons = list("[slot_wear_mask]" = /obj/item/clothing/mask/horsehead/reindeer)
+
+
+/spell/targeted/equip_item/horsemask/christmas/summon_item(var/new_type)
```
Comment (DamianX) [inline code/modules/spells/targeted/equip/horsemask.dm]: I don't think there's a reason to override this proc

**Comment by Arthurdentist** [comment]
@DamianX you are correct, was unnecessary so I removed it.

File: code/game/gamemodes/wizard/spellbook.dm [hunk 2]
```diff
@@ -95,6 +96,10 @@
 		available_artifacts.Remove(T) //Remove the path from the list
 	//Result is a list full of /datum/spellbook_artifact objects
 
+	if(Holiday == "Christmas")
```
Comment (ShiftyRail) [inline code/game/gamemodes/wizard/spellbook.dm:99]: You could do this a bit more OOP if you set a `Holiday_required` for each spell and checked for it when giving out spells.

That way, you can also add Halloween spells without too much hassle, for example.

**Comment by ShiftyRail** [comment]
@Arthurdentist You could use "Holiday required" as a list and do the following : 
```dm
var/list/Holiday_required = list("Christmas", "Christmas Eve")
/* ... */
if (Holiday in Holiday_required || !Holiday_required.len)
     // Do stuff
```

**Review by PJB3005 — APPROVED** [review]
Outside of Shifty's comment about the holiday hardcode, seems good.

**Comment by ShiftyRail** [comment]
@PJB3005 You can merge it as-in, I'll make a second PR to sort the spells because having all the spells being of the same subtypes is frankly a bit dumb

**Comment by Arthurdentist** [comment]
its not ready for merging yet because I might still make a couple more spells, but I talked to @ShiftyRail about the hardcode issue and its something that is probably done better in a second PR (Which ShiftyRail is going to make after this)

**Comment by Blithering** [comment]
Flesh to pine? Turns the victim into a Christmas tree and drops all their items at their feet wrapped as presents.

**Comment by Blithering** [comment]
Summon snakes could also be easily made christmassy by replacing the mob spawned with snowmen

**Comment by Arthurdentist** [comment]
Now added a new spell, Elf Curse, its a christmas-themed version of the clown curse that swaps out the clown clothes for a lovely elf costume, sprited by @Blithering 
elf

There is a weird bug at the moment where the elf uniform isn't showing up with its icon, though there is no reason why this should be the case, @ShiftyRail is going to look into it tomorrow and see if he can find what the issue is

File: code/modules/clothing/under/miscellaneous.dm [hunk 3]
```diff
@@ -715,3 +715,13 @@
 	icon_state = "franksuit"
 	item_state = "franksuit"
 	_color = "franksuit"
+
+/obj/item/clothing/under/elf
+	name = "elf uniform"
+	desc = "A festive suit for a festive elf!"
+	icon_state = "elf"
+	item_state = "elf_s"
```
Comment (ShiftyRail) [inline code/modules/clothing/under/miscellaneous.dm]: There appears to be no such item_state, but `elf` works. Copypaste error, perhaps ?
Same thing for `_color`.

**Comment by SonixApache** [comment]
Santa outfit for wizards?

**Pull request merged**